### DISCUSSION TITLE
How to set a timeout for user ?

### DISCUSSION THREAD

**Discussion by BobSquarePants** [pallets/flask#5092] [Q&A] [answered]
Hi everyone,

in some of my endpoint I'm loading some external .js 😢 

extract of the HTML page rendered by flask/Jinja ->
```html
<!DOCTYPE html>
<html lang="en">

<head>
    <title>A title</title>
    <meta charset="utf-8">  
    <script src="https://something.com/JS/"></script>
</head>
```

But if for any reason the `<script src="https://something.com/JS/"></script>` can't load on the user Internet browser ex: (firewall, something.com offline etc..)

Then the loading of the page is on hold until (I assume) the browser timeout trigger and the rest of the page continue to load. My problem is that meanwhile the server side worker is also on hold 😞 

How could I set a timeout server side ? I'm using Flask <-> Gunicorn <-> Nginx.

Thanks

**Comment by BobSquarePants** [comment]
FYI

I've used also

```html
<script src="https://something.com/JS/" type="module" async></script>
```
and that seem to not block the worker, but anyway I'm still looking for a server side timeout :)

**Comment by ThiefMaster** [comment, marked as answer]
This has nothing to do with Flask/Werkzeug. Also, the request to load the page has already finished by the time JS/CSS are getting loaded.

**Reply by BobSquarePants** [reply]
Thank you @ThiefMaster , yeah it was not seem logic to me either.. but the behavior of the browser loading process is such a mess.. that it mislead me..
